docs(guides) scaffolding guide small rephrasing to improve readability (#3419)
webpack/webpack.js.org: src/content/guides/scaffolding.md

  - EugeneHlushko
---

It can be hard to set up a complex webpack configuration for the first time. Writing advanced configurations to optimize performance could be even more difficult. The `init` feature is designed to support people that want to create their own configuration or initializing projects that other people create.

## Creating a scaffold

Before writing a `webpack-cli` scaffold, think about what you're trying to achieve and who is going to use it:

> - Do you want a generalized scaffold that could be used by any project or type of application?

Reviewer: ```suggestion
- Do you want a generalized scaffold that could be used in wide variety of applications and projects`
```

Author: hmm that makes sense although it should be 
```suggestion
- Do you want a generalized scaffold that could be used by a variety of applications and projects?
```

Author: i think `for wide`  should become `a`

Reviewer: Quantification is important, thus wide should be there.
```suggestion
- Do you want a generic scaffold that could be used by a wide variety of applications and projects?
```

- Do you want something specific, like a scaffold that writes both your `webpack.config.js` and your framework code?
- Who are the potential users and what user experience will look like for the users of your scaffold?

`webpack-cli` offers an interactive experience to customize the output accordingly. For example asking questions like: "What is your entry point?".

### Writing a scaffold

> There are many resources where you can learn how to write a scaffold, you can start from [Writing a Scaffold page](/contribute/writing-a-scaffold/).

Reviewer: ```suggestion
There are many resources where you can learn how to write a scaffold, you can start from [Writing a Scaffold](/contribute/writing-a-scaffold/) guide.
```

Author: This is the writing a scaffold guide though, so it cannot link to another page saying that guide is there :(

Author: on the other hand it says `scaffolding` guide, and the other page is `writing a scaffold` so maybe:
```suggestion
There are many resources where you can learn how to write a scaffold, you can start from reading about [Writing a Scaffold](/contribute/writing-a-scaffold/).
```

Reviewer: Yup, This seems right (Maybe `tutorial` is the word we are looking for?).
Also `There are many resources where you can learn how to write a scaffold` is a bit of an overstatement in my opinion, maybe we can have a better start for this sentence.

Author: Yes, potentially
```suggestion
There are various resources where you can learn how to write a scaffold, you can start by reading [Writing a Scaffold](/contribute/writing-a-scaffold/) tutorial.
```

Reviewer: Let's go ahead with this one
